cpp/velox/substrait/SubstraitToVeloxPlan.cc

```diff
@@ -1743,19 +1749,15 @@ void SubstraitToVeloxPlanConverter::constructSubfieldFilters(
   uint32_t rangeSize = std::max(filterInfo.lowerBounds_.size(), filterInfo.upperBounds_.size());
 
   if constexpr (KIND == facebook::velox::TypeKind::HUGEINT) {
+    // TODO: open here when the Velox's modification is ready
+    VELOX_NYI("constructSubfieldFilters not support for HUGEINT type");
+  } else if constexpr (KIND == facebook::velox::TypeKind::ARRAY || KIND == facebook::velox::TypeKind::MAP) {
+    // Only IsNotNull filter is supported for the above two type kinds now
     if (rangeSize == 0 && !nullAllowed) {
-      // TODO
-      if (fileFormat_ == facebook::velox::dwio::common::FileFormat::ORC) {
-        // uncomment the following line when Velox decimal support IsNotNull filter
-        // filters[common::Subfield(inputName, true)] = std::move(std::make_unique<common::IsNotNull>());
-        VELOX_NYI("constructSubfieldFilters not support IsNotNull for input type '{}'", inputType);
-      } else {
-        VELOX_NYI("constructSubfieldFilters not support IsNotNull for input type '{}'", inputType);
-      }
+      filters[common::Subfield(inputName, true)] = std::move(std::make_unique<common::IsNotNull>());
```

Comment:
> bool nullAllowed = filterInfo.nullAllowed_;

Do we need to obey the nullAllowed info?

Reply:
Maybe a better way to achieve this is to separate the unsupported filters to remaining filter in `separateFilters`. Only filters can be pushed down are sent to `constructSubfieldFilters`.

Reply:
sounds better, but how about keeping this change for this PR, and proposing a new PR to refactor this.